superagi/llms/llm_model_factory.py

```diff
@@ -17,13 +22,33 @@ def get_model(self, model, **kwargs):
 
 
 factory = ModelFactory()
-factory.register_format("gpt-4", lambda **kwargs: OpenAi(model="gpt-4", **kwargs))
-factory.register_format("gpt-4-32k", lambda **kwargs: OpenAi(model="gpt-4-32k", **kwargs))
-factory.register_format("gpt-3.5-turbo-16k", lambda **kwargs: OpenAi(model="gpt-3.5-turbo-16k", **kwargs))
-factory.register_format("gpt-3.5-turbo", lambda **kwargs: OpenAi(model="gpt-3.5-turbo", **kwargs))
-factory.register_format("google-palm-bison-001", lambda **kwargs: GooglePalm(model='models/chat-bison-001', **kwargs))
-factory.register_format("chat-bison-001", lambda **kwargs: GooglePalm(model='models/chat-bison-001', **kwargs))
-
 
-def get_model(api_key, model="gpt-3.5-turbo", **kwargs):
-    return factory.get_model(model, api_key=api_key, **kwargs)
\ No newline at end of file
+def get_model(organisation_id, api_key, model="gpt-3.5-turbo", **kwargs):
+    print("Fetching model details from database...")
+    engine = connect_db()
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    model_instance = session.query(Models).filter(Models.org_id == organisation_id, Models.model_name == model).first()
+    response = session.query(ModelsConfig.source_name).filter(ModelsConfig.org_id == organisation_id,
+                                                                   ModelsConfig.id == model_instance.model_provider_id).first()
+    provider_name = response.source_name
+
+    session.close()
+
+    if provider_name == 'OpenAI':
+        print("Provider is OpenAI")
+        factory.register_format(model_instance.model_name, lambda **kwargs: OpenAi(model=model_instance.model_name, **kwargs))
```

Comment:
no need of factory anymore. u can return the model with params based on input.

Reply:
done